packages/vite/src/executors/dev-server/dev-server.impl.ts

```diff
@@ -37,10 +37,13 @@ export async function* viteDevServerExecutor(
     projectRoot,
     buildTargetOptions.configFile
   );
-  const { serverOptions, otherOptions } = await getServerExtraArgs(options);
+  const { serverOptions, otherOptions } = await getServerExtraArgs({
+    ...options,
+    ...buildTargetOptions,
```

Comment:
So `buildTarget` options should override the `serve` target options? 🤔

Reply:
Yes, because ideally there would not be any overriding done, since there are no common options in serve and build. In any case, it's better to configure your project in `vite.config.ts`!

if this surfaces any errors we can revisit, but we're trying to keep the executor logic as minimal as possible.

Reply:
I see how this may seem confusing or wrong, however here's the thing:

When the options are loaded in the executor, the executor loads the main options from build, along with the options under the `development` configuration, for example:


```
    "build": {
      "executor": "@nx/vite:build",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/apps/my-app",
        "generatePackageJson": false
      },
      "configurations": {
        "development": {
          "mode": "development",
          "watch": true
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "serve": {
      "executor": "@nx/vite:dev-server",
      "options": {
        "buildTarget": "my-app:build"
      },
      "configurations": {
        "development": {
          "buildTarget": "my-app:build:development",
          "hmr": true
        },
        "production": {
          "buildTarget": "my-app:build:production",
          "hmr": false
        }
      }
    },
```

It's up to the user to make sure these are set the way the user prefers. 

In this case, it would load from build the following:

```
{
        "outputPath": "dist/apps/my-app",
        "generatePackageJson": false,
        "mode": "development",
         "watch": true
 }
```

And the options that would be loaded from serve would be:
```
{
          "buildTarget": "my-app:build:development",
          "hmr": true
 }
```

All these will be simplified in the near future, when we will be moving away from executors, and the logic will be much more transparent.

Reply:
I see! Thanks for the taking the time to explain! 🙇🏻‍♂️